src/power_grid_model_io/converters/pandapower_converter.py

```diff
@@ -374,6 +377,36 @@ def _create_pgm_input_links(self):
 
         self.pgm_data["link"] = pgm_links
 
+    def _create_pgm_input_ward(self):  # pragma: no cover
+        # TODO: create unit tests for the function
+        pp_wards = self.pp_data["ward"]
+
+        if pp_wards.empty:
+            return
+
+        if len(pp_wards.index) > 0:
```

Comment:
I guess you don't need this if statement. If pp_wards was empty, you already returned at line 385.

Reply:
But if there are components in the ward dataframe, it raises not implemented. Thats the intention right?

Reply:
Yes, that is the intention. One checks if the df is empty, if it is, returns nothing. The other checks if there is at least one ward, if there is, then raises the error.

Reply:
@Laurynas-Jagutis, we are wrong haha. Line no 387 is redundant since if it is 0, it wouldnt reach that point.